x-pack/filebeat/input/gcs/scheduler.go

```diff
@@ -92,40 +85,54 @@ func (l *limiter) release() {
 	l.wg.Done()
 }
 
-func (s *scheduler) scheduleOnce(ctxWithTimeout context.Context) error {
+func (s *scheduler) scheduleOnce(ctx context.Context) error {
 	defer s.limiter.wait()
-	pager := s.fetchObjectPager(ctxWithTimeout, s.src.MaxWorkers)
+	pager := s.fetchObjectPager(ctx, *s.cfg.MaxWorkers)
+	numObs := 0
+	numJobs := 0
 	for {
 		var objects []*storage.ObjectAttrs
 		nextPageToken, err := pager.NextPage(&objects)
 		if err != nil {
 			return err
 		}
+		numObs += len(objects)
 		jobs := s.createJobs(objects, s.log)
+		s.log.Debugf("scheduler: %d objects fetched for current batch", len(objects))
 
 		// If previous checkpoint was saved then look up starting point for new jobs
 		if !s.state.checkpoint().LatestEntryTime.IsZero() {
 			jobs = s.moveToLastSeenJob(jobs)
 			if len(s.state.checkpoint().FailedJobs) > 0 {
-				jobs = s.addFailedJobs(ctxWithTimeout, jobs)
+				jobs = s.addFailedJobs(ctx, jobs)
 			}
 		}
+		s.log.Debugf("scheduler: %d jobs scheduled for current batch", len(jobs))
 
 		// distributes jobs among workers with the help of a limiter
 		for i, job := range jobs {
+			numJobs++
 			id := fetchJobID(i, s.src.BucketName, job.Name())
 			job := job
 			s.limiter.acquire()
 			go func() {
 				defer s.limiter.release()
-				job.do(s.parentCtx, id)
+				job.do(ctx, id)
 			}()
 		}
 
+		s.log.Debugf("scheduler: total objects read till now: %d", numObs)
+		s.log.Debugf("scheduler: total jobs scheduled till now: %d", numJobs)
+		if len(jobs) > 0 {
+			s.log.Debugf("scheduler: first job in current batch: %s", jobs[0].Name())
+			s.log.Debugf("scheduler: last job in current batch: %s", jobs[len(jobs)-1].Name())
+		}
```

Comment:
These could be bundled into two `s.log.Debugf` calls so that the log level isn't checked 2×.